server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -352,7 +373,14 @@ public void stopThreadResourceTracking(long threadId, ResourceStatsType statsTyp
                     threadResourceInfo.setActive(false);
                     threadResourceInfo.recordResourceUsageMetrics(resourceUsageMetrics);
                     decrementResourceTrackingThreads();
-                    resourceTrackingListeners.forEach(listener -> listener.onTaskExecutionFinishedOnThread(this, threadId));
+                    resourceTrackingListeners.forEach(listener -> {
+                        try {
+                            listener.onTaskExecutionFinishedOnThread(this, threadId);
+                        } catch (Exception e) {
+                            logger.warn("failure in listener during handling of onTaskExecutionFinishedOnThread", e);
```

Comment:
I believe the reason for swallowing the exception here is that if some thread still invokes the listener after task is finished, then we want to ignore that. Is that correct? Can you please add a comment on why the exception is swallowed?

Reply:
Yes, that's right.

These are pluggable listeners which can be used to extend functionality to many different use-cases. We don't want failure in any one of the listener to impact the functionality of other listeners, or the task resource tracking flow in general.